room.go

```diff
@@ -65,9 +72,9 @@ func CreateRoom() *Room {
 	r := &Room{
 		engine:       engine,
 		participants: &sync.Map{},
-		Callback:     NewRoomCallback(),
```

Comment:
since Callback is a private field now, we should take it in as an argument to `CreateRoom`